protocol/x/vault/keeper/orders.go

```diff
@@ -269,51 +280,66 @@ func (k Keeper) GetVaultClobOrders(
 	) *clobtypes.Order {
 		// Ask: leverage_i = leverage - i * order_size_pct
 		// Bid: leverage_i = leverage + i * order_size_pct
-		// skew_i = -leverage_i * spread * skew_factor
 		leveragePpmI := lib.BigU(layer)
 		leveragePpmI.Mul(leveragePpmI, orderSizePctPpm)
 		if side == clobtypes.Order_SIDE_SELL {
 			leveragePpmI.Neg(leveragePpmI)
 		}
 		leveragePpmI.Add(leveragePpmI, leveragePpm)
-		skewPpmI := leveragePpmI.
-			Mul(leveragePpmI, spreadPpm).
-			Mul(leveragePpmI, skewFactorPpm).
-			Quo(leveragePpmI, lib.BigIntOneMillion()).
-			Quo(leveragePpmI, lib.BigIntOneMillion()).
-			Neg(leveragePpmI)
-
-		// spread_i = spread * (layer+1)
-		// negated for buys
-		spreadPpmI := lib.BigU(layer + 1)
-		spreadPpmI.Mul(spreadPpmI, spreadPpm)
-		if side == clobtypes.Order_SIDE_BUY {
-			spreadPpmI.Neg(spreadPpmI)
+
+		// Calculate skew.
+		skewPpmI := lib.BigMulPpm(leveragePpmI, skewFactorPpm, true)
+		if leveragePpm.Sign() < 0 {
+			if side == clobtypes.Order_SIDE_SELL {
+				// ask when short: skew_i = (skew_factor * leverage_i - 1)^2 - 1
+				skewPpmI.Sub(skewPpmI, lib.BigIntOneMillion())
+				skewPpmI = lib.BigMulPpm(skewPpmI, skewPpmI, true)
+				skewPpmI.Sub(skewPpmI, lib.BigIntOneMillion())
+			} else {
+				// bid when short: skew_i = -skew_factor * leverage_i
+				skewPpmI.Neg(skewPpmI)
+			}
+		} else {
+			if side == clobtypes.Order_SIDE_SELL {
+				// ask when long: skew_i = -skew_factor * leverage_i
+				skewPpmI.Neg(skewPpmI)
+			} else {
+				// bid when long: skew_i = -((skew_factor * leverage_i + 1)^2 - 1)
+				skewPpmI.Add(skewPpmI, lib.BigIntOneMillion())
+				skewPpmI = lib.BigMulPpm(skewPpmI, skewPpmI, true)
+				skewPpmI.Sub(skewPpmI, lib.BigIntOneMillion())
+				skewPpmI.Neg(skewPpmI)
+			}
 		}
 
-		// price = oracleprice * (1 + skew_i + spread_i)
-		// price <= oracleprice for buys
-		// price >= oracleprice for sells
-		spreadSkewPpm := new(big.Int).Add(spreadPpmI, skewPpmI)
-		if side == clobtypes.Order_SIDE_SELL && spreadSkewPpm.Sign() < 0 {
-			spreadSkewPpm.SetUint64(0)
-		} else if side == clobtypes.Order_SIDE_BUY && spreadSkewPpm.Sign() > 0 {
-			spreadSkewPpm.SetUint64(0)
+		// Calculate skewed spread.
+		skewedSpreadPpmI := lib.BigIntOneMillion()
+		if side == clobtypes.Order_SIDE_SELL {
+			skewedSpreadPpmI.Add(skewedSpreadPpmI, skewPpmI)
+		} else {
+			skewedSpreadPpmI.Sub(skewPpmI, skewedSpreadPpmI)
```

Comment:
Is this meant to be:
```
skewedSpreadPpmI.Sub(skewedSpreadPpmI, skewPpmI)
```
?

If so, why doesn't a test catch this?

Reply:
intentionally negated this as 
```
a_i = oraclePrice * (1 + ask_spread_i)
b_i = oraclePrice * (1 - bid_spread_i)
```

so i can just do `1 + skewedSpreadPpmI` for both cases